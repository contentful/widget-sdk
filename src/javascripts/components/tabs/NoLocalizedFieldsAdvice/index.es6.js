import React from 'react';
import Advice from 'components/tabs/Advice/index.es6';
import EmptyContentIcon from 'svg/empty-content-model.es6';

export default function NoLocalizedFieldsAdvice() {
  return (
    <Advice data-test-id="no-content-type-advice">
      <Advice.Icon>
        <EmptyContentIcon />
      </Advice.Icon>
      <Advice.Title>There are no localized fields for this locale</Advice.Title>
      <Advice.Description>
        The locale you have chosen does no have any localized fields to display. You can see which
        fields are editable by switching to a new locale, or by toggling into the multiple locale
        display.
      </Advice.Description>
    </Advice>
  );
}
