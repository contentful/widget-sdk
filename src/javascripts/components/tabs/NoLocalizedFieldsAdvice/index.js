import React from 'react';
import PropTypes from 'prop-types';
import Advice from 'components/tabs/Advice';
import TranslationsIcon from 'svg/translations-icon';
import { css } from 'emotion';
import tokens from '@contentful/forma-36-tokens';

const styles = {
  container: css({
    marginTop: tokens.spacingXl
  })
};

const getDescription = localeName => `
  There are no localized fields to translate for ${localeName}. You can
  switch to a different locale using "Translation" in your sidebar.
`;

const NoLocalizedFieldsAdvice = ({ localeName }) => (
  <div className={styles.container}>
    <Advice data-test-id="no-localized-fields-advice">
      <Advice.Icon>
        <TranslationsIcon />
      </Advice.Icon>
      <Advice.Title>There are no fields to translate</Advice.Title>
      <Advice.Description>{getDescription(localeName)}</Advice.Description>
    </Advice>
  </div>
);

NoLocalizedFieldsAdvice.propTypes = {
  localeName: PropTypes.string.isRequired
};

export default NoLocalizedFieldsAdvice;
