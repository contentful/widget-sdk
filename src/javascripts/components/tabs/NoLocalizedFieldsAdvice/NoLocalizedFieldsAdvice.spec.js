import React from 'react';
import Enzyme from 'enzyme';
import NoLocalizedFieldsAdvice from './index';

describe('NoLocalizedFieldsAdvice', () => {
  const props = {
    localeName: 'German (Germany)'
  };

  const render = () => Enzyme.shallow(<NoLocalizedFieldsAdvice {...props} />);

  it('matches snapshot', () => {
    expect(render()).toMatchSnapshot();
  });
});
