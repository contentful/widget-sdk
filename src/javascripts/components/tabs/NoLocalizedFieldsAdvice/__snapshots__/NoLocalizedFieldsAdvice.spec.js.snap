// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoLocalizedFieldsAdvice matches snapshot 1`] = `
.emotion-0 {
  margin-top: 2rem;
}

<div
  className="emotion-0"
>
  <Advice
    data-test-id="no-localized-fields-advice"
  >
    <Component>
      <translations-icon.svg />
    </Component>
    <Component>
      There are no fields to translate
    </Component>
    <Component>
      
  There are no localized fields to translate for German (Germany). You can
  switch to a different locale using "Translation" in your sidebar.

    </Component>
  </Advice>
</div>
`;
